utils/src/main/java/software/amazon/awssdk/utils/async/ByteBufferStoringSubscriber.java

```diff
@@ -175,12 +193,14 @@ public void onSubscribe(Subscription s) {
         storingSubscriber.onSubscribe(new DemandIgnoringSubscription(s));
         subscription = s;
         subscription.request(1);
+        outstandingDemand.incrementAndGet();
```

Comment:
Should probably also be before the call to `request(1)`, similar to in `maybeRequestMore`